port/win32/omrsysinfo.c

```diff
@@ -535,6 +553,23 @@ WIN32_WINNT version constants :
 						}
 					}
 				}
+#if defined(_WIN32_WINNT_WIN10) && (_WIN32_WINNT_MAXVER >= _WIN32_WINNT_WIN10)
+				/* Windows 11 has same major and minor version as 10, but starts at current build number 22000 */
+				if ((defaultTypeName == PPG_si_osType) && !isServerMajorVersion10 && IsWindows10OrGreater()
+					&& (ERROR_SUCCESS == RegQueryValueExA(hKey, CURRENT_BUILD_KEY, NULL, NULL, NULL, &valueSize))) {
+					char currentbuildBuffer[6]; /* CurrentBuild values have been 5 digits so far */
```

Comment:
There should be some extra space so this doesn't overflow and skip the following code if a larger build number occurs in the future.